### PR TITLE
Add 'passthrough-protocol-name' config option

### PR DESCRIPTION
**Pull request by Prouser123**
Adds a `passthrough-protocol-name` config option, ~~also bumps the `config-version` to 4.~~ *not needed*

I set it to use the map option as it seems to me that it is not possible to change the actual protocol version - came up to me as `1.16.0 (Vanilla)`?

Not entirely sure if this is something that would be accepted (although I will be using it at least) but worth a shot! As said in the `config.yml` comment, this is only really useful for java servers that show a custom protocol. (popular real-world example: The Hive using `BeeCord`)

Let me know if you need anything!

### CODE REVIEW

**Comment by DoctorMacc** [comment]
Thanks for the PR!

Seems like a fine addition to have, but as it's a niche thing and doesn't disrupt the current config I don't think we need to bump up the config version.

File: connector/src/main/java/org/geysermc/connector/configuration/GeyserConfiguration.java [hunk 1]
```diff
@@ -99,7 +102,7 @@
         String getAddress();
 
         int getPort();
-
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/configuration/GeyserConfiguration.java:102]: Revert this change for a cleaner commit.

File: connector/src/main/resources/config.yml [hunk 2]
```diff
@@ -50,9 +50,11 @@ floodgate-key-file: public-key.pem
 # Disabling this will prevent command suggestions from being sent and solve freezing for Bedrock clients.
 command-suggestions: true
 
-# The following two options enable "ping passthrough" - the MOTD and/or player count gets retrieved from the Java server.
+# The following three options enable "ping passthrough" - the MOTD and/or player count and/or protocol name gets retrieved from the Java server.
```
Comment (DoctorMacc) [inline connector/src/main/resources/config.yml:53]: Make this `the MOTD, player count and/or protocol name`

File: connector/src/main/resources/config.yml [hunk 3]
```diff
@@ -50,9 +50,11 @@ floodgate-key-file: public-key.pem
...
 # Relay the MOTD from the remote server to Bedrock players.
 passthrough-motd: false
+# Relay the protocol name (e.g. BungeeCord [X.X], Paper 1.X) - only really useful when using a custom protocol name!
```
Comment (DoctorMacc) [inline connector/src/main/resources/config.yml:56]: Add an explanation that it's used for sites like McSrvStatus? Feel free to go on a second comment line to explain.

File: connector/src/main/resources/config.yml [hunk 4]
```diff
@@ -114,7 +116,7 @@ metrics:
   # UUID of server, don't change!
   uuid: generateduuid
 
-# ADVANCED OPTIONS - DO NOT TOUCH UNLESS YOU KNOW WHAT YOU ARE DOING! 
+# ADVANCED OPTIONS - DO NOT TOUCH UNLESS YOU KNOW WHAT YOU ARE DOING!
```
Comment (DoctorMacc) [inline connector/src/main/resources/config.yml:119]: Forgot to catch this - revert this for a cleaner commit.

**Comment by Prouser123** [comment]
@DoctorMacc Thanks! I've made the requested changes now - with the whitespace changes it was because my editor runs code linting on save + commit :P

**Comment by RednedEpic** [comment]
Thanks for the contribution!
